JoinColony/colonyCDapp
# Disable multi sig extension install when colony version is too low

## Description

- Disable the install button of the multisig extension when the colony version is lower than the minimum version (16).
- Set this functionality up for use in other extensions in the future.

## Testing

Before you can test this properly, you need to deploy colony network version 15 in hardhat, and also run the create data script with the previous version. To do this, stop your environment completely, and follow the below steps:

1. Run `npm run dev` as usual.
2. Once this is up and running, start a hardhat console by running `npm run hardhat`
3. Run the following commands in the hardhat console:

```
deployIMWSS = require('./scripts/deployOldUpgradeableVersion.js').deployColonyVersionIMWSS
cn = await artifacts.require("IColonyNetwork").at("0x777760996135F0791E2e1a74aFAa060711197777")
await deployIMWSS(cn)
```

4. Once this is finished (can take a while) run `await cn.getColonyVersionResolver(15)` to ensure version 15 is deployed.
5. Now, you'll create data, but you need to make sure you create data using the previous version. To do this, run `node scripts/create-data.js --usePreviousColonyVersion`
6. Run `npm run frontend` as usual.

Now, you're ready to test!

* Step 1. Load up the extensions page
<img width="1920" alt="Screenshot 2024-08-19 at 16 47 47" src="https://github.com/user-attachments/assets/f973f42b-3a69-48ab-abf3-07f86e37ae7b">

* Step 2. Go into the multi-sig extension details page. You should see the warning banner, and the install button should be disabled.
<img width="1920" alt="Screenshot 2024-08-19 at 16 48 07" src="https://github.com/user-attachments/assets/c0d3e0da-27e1-4ef7-a524-963329d9d7b3">

* Step 3. Click on the Upgrade Colony link in the warning banner. The action sidebar should open with Upgrade Colony as the action type.
<img width="1920" alt="Screenshot 2024-08-19 at 16 48 42" src="https://github.com/user-attachments/assets/a3613e97-1eb8-446b-96a7-c2c053398b31">


## Diffs

**Changes** 🏗

* Disabled to extension install button if the colony version is lower than required for that extension

Resolves #2918
